Da single source tabs

danamansana opened the pull request:
Up to now there have been several sources of responsibility for tracking
tab names and which tab is selected. This commit continues the process
of reducing this problem by 1) eliminating use of tab indices in favor of using tab names 2) eliminating code that duplicated the list of tab names.

Not sure that we should merge this just yet but it could be the basis of further changes

ktp242 commented:
Thanks for doing this! I think first step we can make sure we lint the updates before we make a PR, and also write comments for the functions!

danamansana commented:
I've added descriptive comments for the new functions and fixed most of the linter errors for the files changed by this branch. There are still a few linter issues in the Results component

ktp242 commented on src/app/components/TabItem/TabItem.jsx:
```diff
@@ -151,46 +126,33 @@ class TabItem extends React.Component {
-
-      const option = (
+    tabArray.forEach((tab) => {
+      const name = getNameForFacet(tab.value);
```
The only function of getNameForFacet from TabIndex.js is to convert the input to be a string. However, `tab.value` is a string already.  We probably don't need this function.

danamansana replied:
Yeah this function is a holdover from when this method was more complicated. I'll remove it

ktp242 commented:
I think I can approve this PR! Just add the update to the changelog plz!
